File: src/main/java/io/github/chaosdave34/kitpvp/commands/LoopTabCompleter.java
package io.github.chaosdave34.kitpvp.commands;

import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.command.TabCompleter;
import org.bukkit.util.StringUtil;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class LoopTabCompleter implements TabCompleter {
    @Override
    public @Nullable List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
        if (args.length == 3) {
            List<String> completions = new ArrayList<>();
            StringUtil.copyPartialMatches(args[2], Bukkit.getCommandMap().getKnownCommands().keySet(), completions);
            return completions;
        }

        return Collections.emptyList();
    }
}

File: src/main/kotlin/io/github/chaosdave34/kitpvp/commands/LoopTabCompleter.kt
package io.github.chaosdave34.kitpvp.commands

import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.command.TabCompleter
import org.bukkit.util.StringUtil

class LoopTabCompleter : TabCompleter {
    override fun onTabComplete(sender: CommandSender, command: Command, label: String, args: Array<String>): List<String> {
        val completions: MutableList<String> = mutableListOf()
        if (args.size == 3) {
            StringUtil.copyPartialMatches(args[2], Bukkit.getCommandMap().knownCommands.keys, completions)
            return completions
        }
        return completions
    }
}